docs/subnets/customize-a-subnet.md

```diff
@@ -534,6 +604,74 @@ This example enables the `feeManagerConfig` at the first block with timestamp >=
+```
+
+You can also check activation of precompiles with [`getActivatePrecompilesAt API`](#getActivatePrecompilesAt) and the current chain config with [`getChainConfig API`](#eth_getChainConfig).
+
+Notice that `precompileUpgrades` entry correctly reflects the changes. That's it, your Subnet is all set and the desired upgrades will be activated at the indicated timestamp!
+
+### Trustless Precompile Activation
+
+Precompiles are contracts that has certain features. These features could only be triggered by privileged addresses in previous releases of subnet-evm. For example, the `feeManagerConfig` precompile allows privileged addresses to change the fee structure of the network:
+
+```json
+{
+  "precompileUpgrades": [
+    {
+      "feeManagerConfig": {
+        "blockTimestamp": 1668950000,
+        "adminAddresses": ["0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"]
+      }
+    }
+  ]
+}
+```
+
+In this example, only the address `0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC` is allowed to change the fee structure of the network. The admin address has to intract with the precompile in order to activate it's effect. I.e it needs to send a transaction with relevant fee structure to change it. This is a very powerful feature, but it also means that the network is vulnerable to a single point of failure. If the address `0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC` is compromised, the network is compromised.
+
+In order to make these precompiles more trustless, we have introduced a new way to activate precompiles without admin addresses. This is done by setting relevant precompile initial configuration in either as an upgrade or as a genesis configuration. For example, the `feeManagerConfig` precompile can be configured as follows to change the fee structure on the activation:
+
+```json
+{
+  "precompileUpgrades": [
+    {
+      "feeManagerConfig": {
+        "blockTimestamp": 1668950000,
+        "initialFeeConfig": {
+          "gasLimit": 20000000,
+          "targetBlockRate": 2,
+          "minBaseFee": 1000000000,
+          "targetGas": 100000000,
+          "baseFeeChangeDenominator": 48,
+          "minBlockGasCost": 0,
+          "maxBlockGasCost": 10000000,
+          "blockGasCostStep": 500000
+        }
+      }
+    }
+  ]
+}
+```
+
+Notice that there is no `adminAddresses` field in the configuration. This means that the precompile will be activated with the given configuration without any admin address. It also means there is no need to send a transaction to change the fee structure. The precompile will change the fee at the given timestamp.
+
+This is a trustless way to activate precompiles.
+
+_Note: If you trustlessly activate a precompile, then it will be activated as a precompile. If you want to change the configuration, you will need to first disable it then activate the precompile again with the new configuration._
```

Comment:
Can we simplify this? It's not clear to me what the first sentence is meant to say